webapp/app/js/services/main.js

```diff
@@ -184,12 +184,24 @@ treeherder.factory('thNotify', [
             $log.debug("received message", message);
             var severity = severity || 'info';
             var sticky = sticky || false;
+            var maxNsNotifications = 5;
             thNotify.notifications.push({
                 message: message,
                 severity: severity,
                 sticky: sticky
             });
-            if(!sticky){
+
+            if (!sticky) {
+                if (thNotify.notifications.length > maxNsNotifications) {
+                    /* Replicate shift code to avoid pruning during rapid retriggers
+                     * otheriwse we could use $timeout(thNotify.shift) here */
+                    for(var i=0;i<thNotify.notifications.length; i++) {
```

Comment:
@maurodoglio ya, I had included this explanation in my initial PR description:
*"with many rapid notifications, it appears that $timeout(thNotify.shift) will start to do weird things, pruning the queue where the 4000ms timeout should not have occurred. So I chose to duplicate the shift code, which appears to work perfectly."*

But ya, I will have a look at that `return` that @camd mentions. Thanks!